cpp/textures.cpp

```diff
@@ -127,14 +127,20 @@ NAN_METHOD(texImage2D) {
 	
 	if (info.Length() <= 8 || info[8]->IsNullOrUndefined()) {
 		glTexImage2D(target, level, internalformat, width, height, border, format, type, nullptr);
-		return;
+	} else if (info[8]->IsNumber()) {
+		// In WebGL2 the last parameter can be a byte offset if glBindBuffer()
+		// was called with GL_PIXEL_UNPACK_BUFFER prior to glTexImage2D
+		REQ_INT32_ARG(8, offset);
+		// From https://www.khronos.org/registry/OpenGL-Refpages/es3.0/html/glBindBuffer.xhtml:
+		// "The pointer parameter is interpreted as an offset within the buffer object measured in basic machine units."
+		glTexImage2D(target, level, internalformat, width, height, border, format, type, (void*) offset);
```

Comment:
avoid c-style casts, use `reinterpret_cast`, multiline the call since the line becomes way too long